### PR TITLE
docs(filter): Documented third argument of predicate function

### PR DESCRIPTION
**Pull request by ThomasLandauer**

### CODE REVIEW

**Comment by ThomasLandauer** [comment]
I signed it!

File: src/ng/filter/filter.js [hunk 1]
```diff
@@ -34,8 +34,8 @@
  *     `{name: {first: 'John', last: 'Doe'}}` will **not** be matched by `{name: 'John'}`, but
  *     **will** be matched by `{$: 'John'}`.
  *
- *   - `function(value, index)`: A predicate function can be used to write arbitrary filters. The
- *     function is called for each element of `array`. The final result is an array of those
+ *   - `function(value, index, array)`: A predicate function can be used to write arbitrary filters. The
+ *     function is called for each element of `array`, with 3 arguments beeing passed in: the current element's value; its index in the array; the entire array. The final result is an array of those
```
Comment (gkalpak) [inline src/ng/filter/filter.js]: This line is too long. According to the [coding rules][1], lines should be wrapped at 100 characters.

Nitpicking, I find it too verbose (and the phrase `the current element's value` is confusing imo; just `element` sounds clearer). I would probably prefer something like:

> The function is called for each element of `array`, with the element's index and the array itself as extra arguments.

[1]: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-coding-rules

**Comment by Narretz** [comment]
@ThomasLandauer can you please squash these two commits into one and update the branch (git push -f)?

**Comment by ThomasLandauer** [comment]
Sorry, I just use GitHub through the Website and don't know how to do this :-( You can just go ahead and submit it with your name - or whatever. It's just a minor change.......

**Comment by Narretz** [comment]
Landed as 46b7cf7464ef7f37b3d2919847627e58ff7fdca8

**Pull request closed without merging**